### PR TITLE
Make the extra bright lantern actually flash you.

### PR DESCRIPTION
**Pull request by asperger-sind**
## About the PR 
So you know how the uplink lantern says that it flashes you? It doesn't, it parents off the regular lantern but 2% more bright, this PR remedies that and makes it actually useful, uhh, hopefully.

**Changelog**

:cl: BurninDreamer
- tweak: Make the uplink lantern actually flash you.

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Tools/lantern.yml [hunk 1]
```diff
@@ -42,3 +42,11 @@
       radius: 5
       energy: 10
       color: "#FFC458"
+    - type: Flash
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Objects/Tools/lantern.yml:45]: Just as a note, this has only 5 uses before it burns out, which isn't bad necessarily. balance-wise. Just making you aware

File: Resources/Prototypes/Entities/Objects/Tools/lantern.yml [hunk 2]
```diff
@@ -42,3 +42,11 @@
...
+    - type: MeleeWeapon
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Objects/Tools/lantern.yml:46]: this MeleeWeapon definition is unnecessary and can be deleted entirely.

File: Resources/Prototypes/Entities/Objects/Tools/lantern.yml (hunk 1 above)
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Objects/Tools/lantern.yml:45]: Additionally, in order for the sprite to work, it needs to be given states for "flashing" and "burnt". You could probably just reuse existing sprites for those, as i don't think it matters too much. As it stands right now, it errors when it flashes and also displays an error sprite upon being burnt out.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
Good idea, it just needs a bit of refinement in order to work properly.

File: Resources/Prototypes/Entities/Objects/Tools/lantern.yml (hunk 1 above)
Comment (asperger-sind) [reply Resources/Prototypes/Entities/Objects/Tools/lantern.yml:45]: I'd make it into 10 or something, not really sure how tho.

**Comment by asperger-sind** [comment]
Not sure why the RSI validator is complaining, no idea what it means either, locally everything seems to work fine tho.

File: Resources/Prototypes/Entities/Objects/Tools/lantern.yml (hunk 1 above)
Comment (Peptide90) [reply Resources/Prototypes/Entities/Objects/Tools/lantern.yml:45]: Check out the flash component to see what the datafield is, could be amount:

**Comment by Peptide90** [comment]
> Not sure why the RSI validator is complaining, no idea what it means either, locally everything seems to work fine tho.

RSI and meta.json looks fine, can’t see why it would fail. Just for consistency though I’d probably put the new states above all the inhands ones rather than in the middle, but that’s my OCD.

File: Resources/Prototypes/Entities/Objects/Tools/lantern.yml (hunk 1 above)
Comment (asperger-sind) [reply Resources/Prototypes/Entities/Objects/Tools/lantern.yml:45]: found the datafield, it's "uses:", tried setting it to a different amount, server doesn't run.

Comment (asperger-sind) [reply Resources/Prototypes/Entities/Objects/Tools/lantern.yml:45]: turns out to just be a blank line at the end, everything works after removing it.

**Comment by asperger-sind** [comment]
Checks passed, PR works more or so as intended, ready for merge:tm:

**Review by ZeroDayDaemon — APPROVED** [review]

**Review by EmoGarbage404 — APPROVED** [review]

**Pull request merged**